src/main/java/de/tum/cit/aet/artemis/quiz/service/QuizExerciseImportService.java

```diff
@@ -240,38 +240,67 @@ private void setUpDragAndDropMappingsForImport(DragAndDropQuestion dndQuestion)
         dndQuestion.setCorrectMappings(newDragAndDropMappings);
     }
 
+    /**
+     * Prepares a short answer question for import by creating new instances of spots, solutions, and mappings.
+     * This method creates copies of the spots and solutions, resetting their identifiers and mappings.
+     * It then recreates the correct mappings by linking them to the new spots and solutions using either
+     * persistent IDs or temporary IDs from the original objects.
+     *
+     * @param saQuestion the short answer question to set up for import
+     */
     private void setUpShortAnswerQuestionForImport(ShortAnswerQuestion saQuestion) {
-        List<ShortAnswerSpot> newShortAnswerSpots = new ArrayList<>();
-        for (ShortAnswerSpot shortAnswerSpot : saQuestion.getSpots()) {
-            shortAnswerSpot.setId(null);
-            shortAnswerSpot.setQuestion(saQuestion);
-            shortAnswerSpot.setMappings(new HashSet<>());
-
-            newShortAnswerSpots.add(shortAnswerSpot);
+        HashMap<Long, ShortAnswerSpot> newSpotMap = new HashMap<>();
+        List<ShortAnswerSpot> newSpots = new ArrayList<>();
+        for (ShortAnswerSpot oldSpot : saQuestion.getSpots()) {
+            ShortAnswerSpot newSpot = new ShortAnswerSpot();
+            newSpot.setSpotNr(oldSpot.getSpotNr());
+            newSpot.setWidth(oldSpot.getWidth());
+            newSpot.setInvalid(oldSpot.isInvalid());
+            newSpot.setQuestion(saQuestion);
+            newSpot.setMappings(new HashSet<>());
+            Long key = oldSpot.getId() != null ? oldSpot.getId() : oldSpot.getTempID();
+            newSpotMap.put(key, newSpot);
+            newSpots.add(newSpot);
         }
```

Comment:
I think it would make sene to extract this to a helper method that returns the `HashMap<Long, ShortAnswerSpot> newSpotMap`